src/common/fastopt.cpp

```diff
@@ -88,6 +88,7 @@ template struct As<std::vector<int>>;
 // So, we need an explicit definition for uint64_t. Otherwise, there's a linking error on windows.
 // https://software.intel.com/en-us/articles/size-of-long-integer-type-on-different-architecture-and-os/
 template struct As<std::vector<uint64_t>>;
+template struct As<std::vector<size_t>>; //MacOS support
```

Comment:
Why is that not already covered? Is size_t != uint64_t on MacOS? Please explain this in the line-end comment.

Could you also please have a space after // like everywhere else?

Reply:
I have no idea, to be honest. Both of the fastopt changes are pretty cryptic to me. Without this change, the code fails during linking, because of undefined reference to a certain type. Inspecting the `libmarian.a` revealed that it exported 127 symbols of this type vs a 130 on my linux box. Adding `size_t` made the number of symbols consistent.  Normally, they should be identical, but when compiling marian they are not.

Reply:
"The type size_t is an implementation-defined unsigned integer type that is large enough to contain the size in bytes of any object." https://stackoverflow.com/questions/32021860/c-should-you-size-t-with-a-regular-array

Reply:
Let's remove the //MacOs support comment, and then Resolve this.

Reply:
I have updated this and explained. Having both `<size_t>` and  `<uint64_t>` break linux compilation, so I have wrapped this in an `__APPLE__` tag.

Reply:
Ah, got an answer from our C++ people inside. The solution is easier:

Just add a template specialization for the type that size_t expands to on Mac. Then you can eliminate the `#ifdef`.
```
template struct As<std::vector<uint32_t>>;
```

Reply:
The problem is that `uint64_t` on mac expands to `unsigned long long`, whereas on linux it expands to `unsigned long`. `size_t` on both linux and mac expands to `unsigned long`. As such `unsigned long` and `uint64_t` can't coexist on linux. A solution that would work is to make it:
```
template struct As<std::vector<unsigned long>>;
```
is this acceptable on all OSs? `uint32_t` evaluates to `unsinged int` on mac and linux.

Reply:
If it builds, yes.